src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -157,6 +157,12 @@ class ArrayDeque[A] private(var array: Array[AnyRef], var start: Int, var end: I
     }
   }
 
+  override def reverse = {
+    val r = new ArrayDeque[A](initialSize = size)
+    indices.foreach({i => r.appendAssumingCapacity(this(size - i - 1))})
```

Comment:
Just use a while loop.  There's some chance the JVM will recognize the pattern and find some faster way to do it.  If not, at least it's got the bounds under control so it'll be pretty fast regardless.